docker/Dockerfile.nginx

```diff
@@ -0,0 +1,5 @@
+# This Dockerfile is used to build the nginx image with lua and LuaRocks installed
+FROM fabiocicerchia/nginx-lua:1.27.2-almalinux9.4-20240923
```

Comment:
can we use the original nginx image?

Reply:
I know Nginx doesn't come with Lua by default. But isn't that the reason there's `RUN luarocks install lua-cjson` below? I don't get it, why don't we use official NGINX image and just add the lua package?

Reply:
The official Nginx image is lean and lacks prebuilt modules for non-standard features like Lua. Adding Lua to it typically involves either recompiling Nginx with ngx_http_lua_module or using OpenResty (a Lua-focused Nginx distribution). This process increases the complexity and can introduce potential incompatibilities. The fabiocicerchia/nginx-lua image is specifically designed with Lua modules already integrated and tested. It avoids the manual configuration steps needed to compile and add Lua support into Nginx yourself, which can be tedious and error-prone.